### PR TITLE
[feat] ArraySubsetAssertion now handles arrays-list

### PR DESCRIPTION
**Pull request by nikophil**

### CODE REVIEW

File: tests/Assertion/ArraySubsetAssertionTest.php [hunk 1]
```diff
@@ -65,7 +88,49 @@ public function arrayNotSubsetProvider(): iterable
         yield 'different key does not match' => [['not foo' => 'bar'], ['foo' => 'bar']];
         yield 'different value does not match' => [['foo' => 'not bar'], ['foo' => 'bar']];
         yield 'match is strict' => [['foo' => '0'], ['foo' => 0]];
-        yield 'order in list is important' => [[3, 2, 1], [1, 2, 3]];
```
Comment (nikophil) [inline tests/Assertion/ArraySubsetAssertionTest.php]: this in not true anymore

we can add an option for this (or we can decide order must always be respected), but it would add more complexity to `doEvaluateArrayList()`

Comment (kbond) [reply tests/Assertion/ArraySubsetAssertionTest.php]: To clarify, this now is a valid check as proved above with `yield 'subset can be a list with different order' => [[3, 1], [1, 2, 3]];`.

I'm a bit out of my depth here but isn't an array list a [mathematical set](https://en.wikipedia.org/wiki/Set_(mathematics)) where order doesn't matter: "In a set, all that matters is whether each element is in it or not, so the ordering of the elements in roster notation is irrelevant".

If this is true, then I don't think order or an option to check order should be available in `ArraySubsetAssertion`. You'd need to use the other array/json assertions.

**Review by kbond — COMMENTED** [review]
This is awesome - excellent work! I'm glad we can provide this feature.

It's complex but as we can add tests/fixes as we find edge cases.

File: tests/Assertion/ArraySubsetAssertionTest.php (hunk 1 above)
Comment (nikophil) [reply tests/Assertion/ArraySubsetAssertionTest.php]: this is a strong argument to not support order! I'm closing this
